src/foam/dao/journal/PutJournalEntry.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.dao.journal',
+  name: 'PutJournalEntry',
```

Comment:
Rather than having subclasses of JournalEntry, I would rather have 'cmd' be a property of JournalEntry. This lets us disable 'outputDefaultClasses' and decrease journal sizes.

Reply:
Should command just be a character? Or an enum?

Reply:
I've thought about this approach and ultimately went with this because we can now add features independent of any list of commands that already exist and nothing needs to know about them. This lets projects that don't care about certain features never need to know about this. For example, if project 'foo' needs encryption and project 'bar' doesn't, project 'bar' will never get any code that suggests there's an encryption feature on journal entries. If journal size is a concern, we need a better outputter or a CompressedJournalEntry decorator.